extension.go

```diff
@@ -44,6 +45,13 @@ func (e *ext) setMessage(m Message)       {} // noop
 func (e *ext) setOneOf(o OneOf)           {} // noop
 func (e *ext) setExtendee(m Message)      { e.extendee = m }
 
+func (e *ext) Dependents() []Entity {
+	if len(e.dependentsCache) == 0 {
+		e.dependentsCache = append(e.extendee.Dependents(), e.extendee)
```

Comment:
Doing the append against an unrelated slice is dangerous. Do this instead (likewise throughout):

```suggestion
		e.dependentsCache = []Entity{ e.extendee }
		e.dependentsCache = append(e.dependentsCache, e.extendee.Dependents()...)
```

Also for the conditional, checking nil might be better (this one that's not an issue, but for something like Message, it may have no dependents and would be called redundantly).

Reply:
gotcha on the nil thing, why is it dangerous to append against an unrelated slice?

Reply:
https://play.golang.org/p/IrCwZoeBrpA